websites/C/Coursera/presence.ts

```diff
@@ -0,0 +1,341 @@
+type Functionlize<T> = {
+	[P in keyof T]: () => T[P];
+};
+
+interface Route extends Functionlize<Partial<PresenceData>> {
+	path: RegExp;
+	playback?(): boolean;
+	run?(): PresenceData;
+}
+
+const enum Assets {
+	Logo = "https://i.imgur.com/P4XquqE.png",
+}
+
+enum Settings {
+	TIMESTAMP = "timestamp",
+	BUTTONS = "buttons",
+}
+const { href } = document.location,
+	presence = new Presence({
+		clientId: "1179996601327026227",
+	}),
+	startTimestamp: number = Math.floor(Date.now() / 1000),
+	router = ({ path }: { path: string; presenceData: PresenceData }): Route => {
+		const routes: Route[] = [
+			{
+				path: /^\/$/,
+				details: () => "On Homepage",
+				smallImageKey: () => Assets.Reading,
+				smallImageText: () => "Browsing",
+			},
+
+			{
+				path: /^\/account-profile$/,
+				details: () => " Viewing own Profile",
+				smallImageKey: () => Assets.Reading,
+				smallImageText: () => "Browsing",
+			},
+			{
+				path: /^\/user\/.*$/,
+				details: () => {
+					return `Viewing ${
+						document.querySelector(
+							"[data-testid='profile-photo-invite-section-title']"
+						).textContent
+					}'s Profile`;
+				},
+				smallImageKey: () => Assets.Reading,
+				smallImageText: () => "Browsing",
+				buttons: () => [{ label: "Profile Link", url: href }],
+			},
+			{
+				path: /^\/learn\/(?!.*\bhome\b).*\/$/,
+				smallImageKey: () => Assets.Reading,
+				smallImageText: () => "Learning",
+				state: () => {
+					return `Subject: ${document.title.substring(
+						0,
+						document.title.lastIndexOf(" ") - 1
+					)}`;
+				},
+
+				details: () => {
+					return href.split("/")[4].split("-").join(" ");
+				},
+				buttons: () => [
+					{
+						label: "View Course",
+						url: (() => {
+							return href.split("/").slice(0, 5).join("/");
+						})(),
+					},
+				],
+			},
+			{
+				path: /^\/learn\/.*\/home\/week\/.*$/,
+				smallImageKey: () => Assets.Reading,
+				smallImageText: () => "Learning",
+				state: () => {
+					return href.split("/")[4].split("-").join(" ");
+				},
+				details: () => {
+					return `Viewing : ${href
+						.split("/")[6]
+						.split("-")
+						.join(" ")
+						.toUpperCase()} ${href.split("/")[7]}'s content`;
+				},
+				buttons: () => [
+					{
+						label: "View Course",
+						url: (() => {
+							return href.split("/").slice(0, 5).join("/");
+						})(),
+					},
+				],
+			},
+			{
+				path: /^\/learn\/.*\/home\/(?!.*\bweek\b).*$/,
+				smallImageKey: () => Assets.Reading,
+				smallImageText: () => "Learning",
+				state: () => {
+					return `Viewing Course ${href.split("/")[6].split("-").join(" ")}`;
+				},
+				details: () => {
+					return href.split("/")[4].split("-").join(" ");
+				},
+				buttons: () => [
+					{
+						label: "View Course",
+						url: (() => {
+							return href.split("/").slice(0, 5).join("/");
+						})(),
```

Comment:
Ditto (and for others like this)
```suggestion
						url: href.split("/").slice(0, 5).join("/"),
```